src/test/java/org/apache/commons/lang3/StringUtilsTest.java

```diff
@@ -2687,6 +2687,9 @@ public void testRemoveIgnoreCase_String() {
 
         // StringUtils.removeIgnoreCase("queued", "zZ") = "queued"
         assertEquals("queued", StringUtils.removeIgnoreCase("queued", "zZ"));
+
+        // StringUtils.removeIgnoreCase("İa", "a") = "İ"
+        assertEquals("İ", StringUtils.removeIgnoreCase("İa", "a"));
```

Comment:
We need to minimally replace `"İ"` with `"\u0130"` for this to properly work. For example try running:

```java
System.out.println("İ");
System.out.println("\u0130");
```
only the second will give your desired output. That said, your point is still quite valid.